src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -24,6 +24,8 @@ trait Ledger {
 
 class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with Logger {
 
+  val rewardTeller = new RewardTeller(blockchainConfig.monetaryPolicyConfig)
```

Comment:
I see your point, and yeah I wouldn't change it at this time. Currently the dependency between `Ledger` and `RewardTeller` feels like natural parent-child one, also reflected in the config.